### DISCUSSION TITLE
Would consts for sequence strings be a good idea?

### DISCUSSION THREAD

**Discussion by BloodyRain2k** [alejandroautalan/pygubu-designer#216] [Ideas]
There's one thing that I've so far found a little irritating with using tkinter, which is the sequence strings.
I understand that they're strings because as the name implies, they're not simple a 1-off name for something, so the choice for strings makes sense.

But I think in case of "custom" events the case is a little different. In my example the `CalendarFrame`'s `<<CalendarFrameDateSelected>>` event.
Typing that one out and not making a typo is something, so I was wondering if it might be an idea to have such "fixed" event names as consts in the classes they belong to instead.

I've done a quick test here https://github.com/alejandroautalan/pygubu/compare/master...BloodyRain2k:pygubu:sequence_const as example and to see if it'd even work, and from some quick testing it does seem to work.

But I'm primarily looking for feedback of everyone else here. It could be quite an effort to convert all events that pygubu adds, and if I'm the only one who even cares about that, then there'd be little point in making an effort over just converting each as I run into them.

**Comment by alejandroautalan** [comment]
Hello,

> But I think in case of "custom" events the case is a little different. In my example the CalendarFrame's <<CalendarFrameDateSelected>> event.
> Typing that one out and not making a typo is something, so I was wondering if it might be an idea to have such "fixed" event names as consts in the classes they belong to instead.

I think that this is a good idea for widget custom events (or virtual events).

As for the name of the constant in the class, I would like to be prefixed with "Event", something like:

```python
# ...
    EventCalendarFrameDateSelected = "<<CalendarFrameDateSelected>>"
# ...
    virtual_events = (CalendarFrame.EventCalendarFrameDateSelected,)
# ...
    self.event_generate(CalendarFrame.EventCalendarFrameDateSelected)
```

Let's see what other people thinks.

Regards
Alejandro A.

**Reply by BloodyRain2k** [reply]
With the naming I went with `SEQ_` for a few reasons:

While they /are/ events, Tk refers to them left and right as 'sequences', so 'event' /might/ feel out of place.
Then again, everyone familiar with Tk knows that 'sequences' are events.

Constants in Python are normally all caps, which makes longer names awful to read.
Or need a lot of underscores. That's why I capitalized the `SEQ_` part, to hopefully hint at it being a const, while leaving everything else 'normal' for readability.

But yeah, let's see what others think. Wouldn't be the first time that my view of something making sense only makes sense to me xD

**Reply by jrezai** [reply]
I'd be fine with either regular strings or _SEQ__.
